pkg/upgrade/upgrade.go

```diff
@@ -415,3 +459,36 @@ func getKfDefInstances(c client.Client) (*kfdefv1.KfDefList, error) {
 
 	return kfDefList, nil
 }
+
+func deleteDeployments(cli client.Client, applicationNamespace string) error {
+	// In v2, Deployment selectors use a label "app.opendatahub.io/<componentName>" which is
+	// not present in v1. Since label selectors are immutable, we need to delete the existing
+	// deployments and recreated them.
+
+	// Delete Deployment objects
+
+	var multiErr *multierror.Error
+	deployments := &appsv1.DeploymentList{}
+	listOpts := &client.ListOptions{
+		Namespace: applicationNamespace,
+	}
+	err := cli.List(context.TODO(), deployments, listOpts)
+	if err != nil {
+		return err
+	}
+	// filter deployment which has the new label to limit that we do not over kill other deployment
+	// this logic can be used even when upgrade from v2.4 to v2.5 without remove it
+	for _, deployment := range deployments.Items {
+		// fmt.Println("start with deployment " + deployment.Name)
+		selectorLabels := deployment.Spec.Selector.MatchLabels
+		for label := range selectorLabels {
+			if strings.Contains(label, "app.opendatahub.io/") {
+				// this deployment has the new label, this is a v2 to v2 upgrade
+				continue
+			}
+		}
+		multiErr = multierror.Append(multiErr, cli.Delete(context.TODO(), &deployment, []client.DeleteOption{}...))
```

Comment:
nit: `[]client.DeleteOption{}...` not needed here
```suggestion
		multiErr = multierror.Append(multiErr, cli.Delete(context.TODO(), &deployment))
```